File: src/bot/listeners/contents/videoMessageHandler.ts
import { Client } from 'tdl';
import { updateNewMessage } from 'tdlib-types';
import { VIDEO_MESSAGE_DESTINATION } from '../../../constants';
import { refreshChats } from '../../helper';

export const videoMessageHandler = async (
  update: updateNewMessage,
  client: Client
): Promise<void> => {
  console.log('Received video message');

  // get chats
  const { topic } = await refreshChats(client, update.message.chat_id, {
    chatId: VIDEO_MESSAGE_DESTINATION.chat,
    topicName: VIDEO_MESSAGE_DESTINATION.thread,
  });

  // forward files
  await client.invoke({
    _: 'forwardMessages',
    chat_id: VIDEO_MESSAGE_DESTINATION.chat,
    message_thread_id: topic?.info.message_thread_id,
    message_ids: [update.message.id],
    from_chat_id: update.message.chat_id,
  });
};

File: src/main/java/com/pirasalbe/services/telegram/listeners/contents/VideoMessageHandler.java
package com.pirasalbe.services.telegram.listeners.contents;

import java.util.function.BiConsumer;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import com.pirasalbe.configurations.TelegramConfiguration;
import com.pirasalbe.helpers.TelegramBotHelper;
import com.pirasalbe.helpers.models.ForumInfo;
import com.pirasalbe.helpers.models.RefreshResult;

import it.tdlight.client.SimpleTelegramClient;
import it.tdlight.jni.TdApi.ForwardMessages;
import it.tdlight.jni.TdApi.UpdateNewMessage;

@Component
public class VideoMessageHandler implements BiConsumer<SimpleTelegramClient, UpdateNewMessage> {

	private static final Logger LOGGER = LoggerFactory.getLogger(VideoMessageHandler.class);

	@Autowired
	private TelegramConfiguration configuration;

	@Override
	public void accept(SimpleTelegramClient client, UpdateNewMessage update) {
		LOGGER.info("Received new video message");

		// get chats
		RefreshResult refreshResult = TelegramBotHelper.refreshChats(client, update.message.chatId,
				new ForumInfo(configuration.getVideoMessageDestinationChatId(),
						configuration.getVideoMessageDestinationThreadName()));

		TelegramBotHelper.sendSync(client,
				new ForwardMessages(configuration.getVideoMessageDestinationChatId(),
						refreshResult.getTopic().map(topic -> topic.info.messageThreadId).orElse(0l),
						update.message.chatId, new long[] { update.message.id }, null, false, false));
	}

}
